packages/sdk-codegen/src/sdkModels.ts

```diff
@@ -1560,7 +1567,13 @@ export class ArrayType extends Type {
 export class EnumType extends Type implements IEnumType {
   readonly values: EnumValueType[]
 
-  constructor(public elementType: IType, schema: OAS.SchemaObject) {
+  constructor(
+    public elementType: IType,
+    schema: OAS.SchemaObject,
+    types: TypeList,
+    typeName?: string,
+    methodName?: string
+  ) {
```

Comment:
I'm happy to discuss the enum naming logic. I put it in the place where the enum was added to the model because that's where it was needed. I don't know how `registerEnum()` would be different than `makeEnum()`, but again, let's talk about it.

Reply:
maybe it's just how my brain works but I was reading the `new EnumType(...)` as "here's the enum in all it's entirety" and my brain glanced over the `registerEnum` call because I didn't think that would have a side effect on the enum object itself. So it took me a little while to figure out that it was changing the name. I didn't think I'd need to read the contents of the `registerEnum` method to learn about how properties of the enum instance itself were set. I'd expect that to be found more in either the constructor or a method with a more aggressively mutating sounding name :-)